### ISSUE TITLE
@dxos/client package is not working with Bun

### ISSUE THREAD

**Issue by ansarizafar** [oven-sh/bun#3805]
### What version of Bun is running?

0.7.0

### What platform is your computer?

Linux 5.19.0-46-generic x86_64 x86_64

### What steps can reproduce the bug?

I am getting Segmentation fault (core dumped) error when I try to run index.ts

```shell
bun add @dxos/client
```
```javascript
import { Client } from '@dxos/client';

// create a client
const client = new Client();

await client.initialize();
````
```shell
bun run index.ts
```

### What is the expected behavior?

Script should not exit with Segmentation fault (core dumped) error.

### What do you see instead?

Segmentation fault (core dumped)

**Comment by zhenyasav** [comment]
this was likely attempted with `@dxos/client@0.1.53`

**Comment by Jarred-Sumner** [comment]
This has been fixed as of Bun v0.7.3. Now it throws an AssertionError.

```js
~/Desktop is 📦 v0.1.0 via 🐹 v1.20.3
❯ bun-0.7.1 seggy.js
fish: Job 1, 'bun-0.7.1 seggy.js' terminated by signal SIGSEGV (Address boundary error)

~/Desktop is 📦 v0.1.0 via 🐹 v1.20.3
❯ bun-0.7.3 seggy.js
P2P network is not configured.
2541 |     AssertionError.prototype.name = "AssertionError";
2542 |     function assert(t, m) {
2543 |       if (!t) {
2544 |         var err = new AssertionError(m);
2545 |         if (Error.captureStackTrace)
2546 |           Error.captureStackTrace(err, assert);
              ^
error: AssertionError
      at assert (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2546:10)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/cipher.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2768:4)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/cipher-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2833:17)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/symmetric-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3097:22)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/handshake-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3227:25)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/index.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3653:22)

~/Desktop is 📦 v0.1.0 via 🐹 v1.20.3
❯ bun seggy.js
P2P network is not configured.
2541 |     AssertionError.prototype.name = "AssertionError";
2542 |     function assert(t, m) {
2543 |       if (!t) {
2544 |         var err = new AssertionError(m);
2545 |         if (Error.captureStackTrace)
2546 |           Error.captureStackTrace(err, assert);
              ^
error: AssertionError
      at assert (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2546:10)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/cipher.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2768:4)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/cipher-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:2833:17)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/symmetric-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3097:22)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/handshake-state.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3227:25)
      at __require (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:9:9)
      at node_modules/.pnpm/noise-protocol@3.0.1/node_modules/noise-protocol/index.js (/Users/jarred/Desktop/node_modules/@dxos/hypercore/dist/lib/node/index.cjs:3653:22)
```
